### ISSUE TITLE
Forms completely broken after duplication

### ISSUE THREAD

**Issue by timoschwarzer** [verbb/formie#1027]
### Describe the bug

We created a rather complex form (many conditions for fields) and wanted to duplicate it to translate it into another language. Since the duplication both forms, the original one and the copy are not accessible anymore from both the backend and the frontend. Both crash with the following error:

`
2022-08-03 12:33:17 [web.ERROR] [yii\base\InvalidConfigException] Invalid field layout ID: 73 {"trace":["#0 /app/vendor/verbb/formie/src/elements/Form.php(424): verbb\\formie\\behaviors\\FieldLayoutBehavior->getFieldLayout()","#1 /app/vendor/verbb/formie/src/elements/Form.php(1137): verbb\\formie\\elements\\Form->getFormFieldLayout()","#2 /app/vendor/verbb/formie/src/elements/Form.php(1688): verbb\\formie\\elements\\Form->getCustomFields()","#3 /app/vendor/verbb/formie/src/services/Rendering.php(92): verbb\\formie\\elements\\Form->getFrontEndJsVariables()","#4 /app/vendor/verbb/formie/src/variables/Formie.php(114): verbb\\formie\\services\\Rendering->renderForm(Object(verbb\\formie\\elements\\Form), Array)","#5 /app/vendor/twig/twig/src/Extension/CoreExtension.php(1607): verbb\\formie\\variables\\Formie->renderForm(Object(verbb\\formie\\elements\\Form))","#6 /app/vendor/craftcms/cms/src/helpers/Template.php(101): twig_get_attribute(Object(craft\\web\\twig\\Environment), Object(Twig\\Source), Object(verbb\\formie\\variables\\Formie), 'renderForm', Array, 'method', false, false)","#7 /app/storage/runtime/compiled_templates/3c/3c9135cafe1b63e1f6332fdf5d301df6.php(581): craft\\helpers\\Template::attribute(Object(craft\\web\\twig\\Environment), Object(Twig\\Source), Object(verbb\\formie\\variables\\Formie), 'renderForm', Array, 'method')","#8 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_76c482783440f9f792f2e4ab76e55d13->doDisplay(Array, Array)","#9 /app/vendor/twig/twig/src/Template.php(367): Twig\\Template->displayWithErrorHandling(Array, Array)","#10 /app/storage/runtime/compiled_templates/ca/ca9db13c2468e0abcd59bd8977754b7a.php(110): Twig\\Template->display(Array)","#11 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_468eb260ee289dff08b800622a3c1d93->doDisplay(Array, Array)","#12 /app/vendor/twig/twig/src/Template.php(367): Twig\\Template->displayWithErrorHandling(Array, Array)","#13 /app/storage/runtime/compiled_templates/df/df7bff15b0f24a74df347682132a81d4.php(226): Twig\\Template->display(Array)","#14 /app/vendor/twig/twig/src/Template.php(171): __TwigTemplate_2560dadb29fe97add242f0a97d6e9590->block_main(Array, Array)","#15 /app/storage/runtime/compiled_templates/37/372c3d21fcf92a5aab6eaa8fb029668d.php(219): Twig\\Template->displayBlock('main', Array, Array)","#16 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_46e3e7c52efb4d3d89e1add28f87bc9f->doDisplay(Array, Array)","#17 /app/vendor/twig/twig/src/Template.php(367): Twig\\Template->displayWithErrorHandling(Array, Array)","#18 /app/storage/runtime/compiled_templates/df/df7bff15b0f24a74df347682132a81d4.php(48): Twig\\Template->display(Array, Array)","#19 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_2560dadb29fe97add242f0a97d6e9590->doDisplay(Array, Array)","#20 /app/vendor/twig/twig/src/Template.php(367): Twig\\Template->displayWithErrorHandling(Array, Array)","#21 /app/vendor/twig/twig/src/Template.php(379): Twig\\Template->display(Array)","#22 /app/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\\Template->render(Array, Array)","#23 /app/vendor/twig/twig/src/Environment.php(277): Twig\\TemplateWrapper->render(Array)","#24 /app/vendor/craftcms/cms/src/web/View.php(428): Twig\\Environment->render('stwon/page', Array)","#25 /app/vendor/craftcms/cms/src/web/View.php(481): craft\\web\\View->renderTemplate('stwon/page', Array)","#26 /app/vendor/craftcms/cms/src/web/TemplateResponseFormatter.php(52): craft\\web\\View->renderPageTemplate('stwon/page', Array, 'site')","#27 /app/vendor/yiisoft/yii2/web/Response.php(1095): craft\\web\\TemplateResponseFormatter->format(Object(craft\\web\\Response))","#28 /app/vendor/craftcms/cms/src/web/Response.php(286): yii\\web\\Response->prepare()","#29 /app/vendor/yiisoft/yii2/web/Response.php(339): craft\\web\\Response->prepare()","#30 /app/vendor/yiisoft/yii2/base/Application.php(390): yii\\web\\Response->send()","#31 /app/web/index.php(12): yii\\base\\Application->run()","#32 {main}"],"memory":6819912,"exception":"[object] (yii\\base\\InvalidConfigException(code: 0): Invalid field layout ID: 73 at /app/vendor/verbb/formie/src/behaviors/FieldLayoutBehavior.php:54)"}
`

It is not even possible to delete the forms now.

### Steps to reproduce

I don't know the exact steps to reproduce this issue yet, but it's something along the lines "Create a form, ??? and duplicate it"

### Form settings

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.2.0.2

### Plugin version

2.0.4

### Multi-site?

Yes

**Comment by timoschwarzer** [comment]
What I've found so far is that duplicating a form does not duplicate its field layout.

So first you create a Form with the ID 1 and Field Layout ID 200.
When you duplicate it, there will be a Form ID 2 which also has the Field Layout ID 200. (This alone leads to bugs such as that it will display all form fields twice in the editor, making it impossible to edit the form)
If you now delete one of the two forms, it will delete the Field Layout 200 and thus break the remaining form as well.

This is a pretty heavy and breaking bug introducing data loss and leaving the application in a broken state if you don't know how to fix it by going into the database directly. I'd appreciate if it could be fixed soon.

**Comment by engram-design** [comment]
There is indeed a bug with duplicating forms, that's fixed for the next release. For now, you'll need to "Save as a new form" in order to duplicate a form, until that's released. This was only introduced in 2.0.4 where we added the Duplicate element action.

To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.4",
  "...": "..."
}
```

Then run `composer update`.

**Comment by timoschwarzer** [comment]
@engram-design thank you very much for the quick fix! 🙌

**Comment by engram-design** [comment]
Fixed in [2.0.5](https://github.com/verbb/formie/releases/tag/2.0.5)
